packages/@react-spectrum/s2/chromatic/ActionButton.stories.tsx

```diff
@@ -0,0 +1,112 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {ActionButton, Text} from '../src';
+import {generatePowerset} from '@react-spectrum/story-utils';
+import type {Meta} from '@storybook/react';
+import NewIcon from '../s2wf-icons/S2_Icon_New_20_N.svg';
+import {StaticColorProvider} from '../stories/utils';
+import {style} from '../style/spectrum-theme' with { type: 'macro' };
+
+const meta: Meta<typeof ActionButton> = {
+  component: ActionButton,
+  parameters: {
+    chromaticProvider: {disableAnimations: true}
+  },
+  title: 'S2 Chromatic/ActionButton'
+};
+
+export default meta;
+
+let states = [
+  {isQuiet: true},
+  {isReadOnly: true},
+  {isDisabled: true},
+  {size: ['XS', 'S', 'M', 'L', 'XL']},
+  {staticColor: ['black', 'white']}
+];
+
+let combinations = generatePowerset(states);
+
+function shortName(key, value) {
+  let returnVal = '';
+  switch (key) {
+    case 'isQuiet':
+      returnVal = 'q';
+      break;
+    case 'isReadOnly':
+      returnVal = 'ro';
+      break;
+    case 'isDisabled':
+      returnVal = 'd';
+      break;
+    case 'size':
+      returnVal = `size: ${value}`;
+      break;
+    case 'staticColor':
+      returnVal = `static: ${value}`;
+      break;
+
+  }
+  return returnVal;
+}
+
+const Template = (args) => {
+  let {children, ...otherArgs} = args;
+  return (
+    <div className={style({display: 'grid', gridTemplateColumns: 'repeat(4, minmax(0, 250px))', gridAutoFlow: 'row', justifyItems: 'start', gap: 24, width: '[100vw]'})}>
+      {combinations.map(c => {
+        let key = Object.keys(c).map(k => shortName(k, c[k])).join(' ');
+        if (!key) {
+          key = 'default';
+        }
+
+        let button = <ActionButton key={key} {...otherArgs} {...c}>{children ? children : key}</ActionButton>;
+        if (c.staticColor != null) {
+          return (
+            <StaticColorProvider staticColor={c.staticColor}>
+              {button}
+            </StaticColorProvider>
+          );
+        }
+
+        return button;
+      })}
+    </div>
+  );
+};
+
+export const Default = {
+  render: Template
+};
+
+export const WithIcon = {
+  render: Template,
+  args: {
+    children: <><NewIcon /><Text>Press me</Text></>
+  }
+};
+
+export const IconOnly = {
+  render: Template,
+  args: {
+    children: <NewIcon />
+  }
+};
+
+// TODO: eventually make all none S2 chromatic stories have
+// chromatic: {
+//   disableSnapshot: true
+// }
+// so that we centralize the chromatic stories. Wait until all these chromatic stories are finished
+// and we run chromatic one last time to ensure a good baseline, then have all the stories in the
+// chromatic folder import the original stories and rexport them
```

Comment:
As we add powerset stories, some/all of the currently snapshotted S2 stories are redundant/extraneous so we'll want to centralize them all into this folder eventually